quill-jdbc-monix/src/test/scala/io/getquill/examples/BlowUpDatabaseExample.scala

```diff
@@ -0,0 +1,67 @@
+package io.getquill.examples
+
+import java.sql.{ Connection, ResultSet }
+
+import io.getquill.context.jdbc.Runner
+import io.getquill.util.LoadConfig
+import io.getquill.{ JdbcContextConfig, Literal, PostgresAsyncJdbcContext }
+import monix.execution.Scheduler
+import monix.execution.schedulers.CanBlock
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+/**
+ * This is a long-running test that will cause a OutOfMemory exception if
+ * a ResultSet is not streamed correctly (e.g. if the ResultSet.TYPE_SCROLL_SENSITIVE option
+ * is used which will force most databases to put the entire ResultSet into memory).
+ * Run with -Xmx200m and doBlowUp=true to correctly reproduce the error.
+ */
+object BlowUpDatabaseExample {
```

Comment:
Is this the only thing that would fail if the streaming is not correct? Could it be transformed into a unit test that forks a jvm with a small heap?

Reply:
This 'test' usually takes 5 mintues (to load the database with enough needed values) and the `-Xmx` settings need to be just-right for it to work. I'd love to have some kind of long-running test suite in general but we need many changes in our build infrastucture to do that.

Reply:
Maybe it could use something like this to generate the data? https://stackoverflow.com/questions/24841142/how-can-i-generate-big-data-sample-for-postgresql-using-generate-series-and-rand

Reply:
Yeah, that will be faster so I'll try it. How do we guarentee that the heap will be a certain size though? I'm not sure we can test the negative scenario of this situation (i.e. when the DB **does** blow up). Even if we fork the JVM to do this, I don't think the test will be able to catch/expect an out-of-memory exception.

Reply:
Btw. The AsyncJdbcContextSpec, and PeopleAsyncJdbcSpec also have tests for streaming cases (normally and inside a transaction). BlowUpDatabaseExample is specifically for instances where (some) databases will return the whole resultset into memory as opposed to streaming it when `prepareStatement` is created with something other then `ResultSet.TYPE_FORWARD_ONLY` (e.g. `TYPE_SCROLL_SENSITIVE`) in order to allow the user to scroll it backward.

Reply:
I'm not sure I see the limitation, doesn't this test always fail with a low heap? We can set the heap size when forking the JVM

Reply:
Nope. Even for low heap (e.g. `-Xmx200m` or less) it will still succeed because ResultSet will only materialize 100 objects at a time (since `fetchSize=100`). The `Observable` will stream them out one by one and log them to console. Then they will be garbage-collected when needed. Only if you force the database to load the entire ResultSet into memory (such as when there's a rewindable cursor) will the JVM blow up.